DEVELOPMENT.md

```diff
@@ -0,0 +1,229 @@
+  within the scope of the pull request.
+- Changed codepaths should be exercised by unit tests that cover at least one
+  nominal case and one error case.
+- Pull requests may only be merged with a green build. Do not merge with
+  build failures, even if they are known and unrelated.
+- Flaky or otherwise spurious CI failures count as hard failures. Commit fixes
+  or skips for flaky tests or other failures first!
+
+### Scoping pull requests
+
+Significantly inspired by Phabricator's article about
+[Recommendations on Revision Control][phab-one-idea], and particularly
+because pull requests tend to be squash-merged, try to keep PRs focused
+on one idea (or the minimal number of ideas for the change to be viable).
+
+Some of the advantages of smaller PRs are:
+- quicker to write + quicker to review = faster iteration
+- easier to spot errors
+- less cognitive overhead (and net time invested) for reviewers due to
+  reduced scope
+- naturally avoids scope creep
+- clearly establishes that all parts of the PR are related
+
+Because of this, contributors are encouraged to keep PRs as small as
+possible. Given that this does introduce some developer overhead (e.g.
+needing to manage more PRs), *how* small is unspecified; reviewers
+can request PRs to be broken down further as necessary, and contributors
+should work with reviewers to find the right balance.
+
+[phab-one-idea]: https://secure.phabricator.com/book/phabflavor/article/recommendations_on_revision_control/#one-idea-is-one-commit
+
+## Updating the CHANGELOG
+
+After a pull request is merged, significant changes must be summarized in the
+[CHANGELOG][CHANGELOG.md]. Since the PR number is not known ahead of time, this
+must be done as a subsquent commit after the merged PR (though the CHANGELOG
+PR can be prepared in tandem with the referenced PR).
+
+The format of the CHANGELOG entry should be:
+
+```
+- TYPE **COMPONENT**: Description (#PR_NUMBER)
+```
+
+The `TYPE` should be ommitted if it is a feature. If the change is not a
+feature, `TYPE` should be either `FIX` or `PERF`. New types of changes should
+be added to this document before used if they cannot be categorized by the
+existing types.
+
+New CHANGELOG entries should be added to the current "unreleased" section at
+the top of the CHANGELOG as long as they are within the scope of that potential
+version. If no unreleased section exists, one should be added using the
+appropriate proposed semver. If an unreleased section exists but the new change
+requires a different semver change (e.g. the unreleased version is a patch
+version bump, but the new change requires a minor version bump), the version
+on the existing section should be updated. See [COMPATIBILITY][COMPATIBILITY.md]
+for more information on versioning.
+
+An example CHANGELOG addition:
+
+```
+# 0.4.5 (unreleased)
+
+- FIX **DB**: Index data race in FST Segment reads (#938)
+```
+
+## Building Docs
```

Comment:
OK, I need to update this whole section… Leave it for now, I will create a seperate readme in that directory.